src/pages/snippets/disable-components-using-directive.mdx

```diff
@@ -0,0 +1,53 @@
+---
+title: Disable components using directive
+description: "Example of directive to disable interactive components when a certain condition is met."
+tags: ["angular", "directives"]
+pubDate: Feb 25, 2023
+contributedBy: "@pawelkubiakdev"
+---
+
+The code snippet below shows how to disable (angular material) components if the shouldDisable$ stream from the ExampleService emits true.
+This can be a useful solution when we want to block actions when the user does not have the appropriate permissions resulting from:
+- user role
+- a feature that is blocked for the user etc.
+
+
+```typescript
+@Directive({
+  selector: '[disableInteractiveElements]',
+  standalone: true,
+  hostDirectives: [DestroyedDirective]
+})
+export class DisableInteractiveElementsDirective implements OnInit {
+  private readonly destroyed$ = inject(DestroyedDirective).destroyed$;
+
+  constructor(private readonly service: ExampleService,
+              @Optional() @Self() private readonly button: MatButton,
```

Comment:
I think you should be able to make this more generic by doing something like

```
private readonly elementRef: ElementRef
```

as that will get the generic host element

then you could do something like

```typescript
    this.service.shouldDisable$
    .pipe(takeUntil(this.destroyed$))
    .subscribe((shouldDisable) => {
      if (Object.hasOwnProperty(this.elementRef)) {
        this.elementRef.disabled = shouldDisable;
      } 
    });
```

Then it would work for any host element that has a disabled property.

Reply:
Good idea, to make it more generic. Will adjust it.

Reply:
I'd give it a try in a sample app. I thiiiink it will work but not positive.

Reply:
Hi guys,
@santoshyadavdev - I've added the link to the tweet about the destroyed directive.
@yharaskrik - I remembered why I once implemented it this way. Using elementRef.nativeElement.disabled = true, it doesn't work for `MatSelect`. 
For `MatButton` it works in most cases, i.e. it works when we trigger the change of the enabled/disabled status change in runtime, but it doesn't work when we want the button to be disabled on init. At first glance, it looks like it's some kind of problem with change detection on the material side.

I added elementRef to the code snippet so that we can disable other elements that will allow us to do so, e.g. input, html select, or button.

Reply:
Approved @yharaskrik let me know if it looks good now?

Reply:
Ya looks good to me